File: client/source/android/src/com/hungkv/autolikeapp/database/Transaction.java
package com.hungkv.autolikeapp.database;

public class Transaction {
    private int id;
    private String phone;
    private String code;
    private int value;
    private String time;
    private String updateTime;
    private boolean status;

    public Transaction(int id, String phone, String code, int value, String time, String updateTime, boolean status) {
        this.id = id;
        this.phone = phone;
        this.code = code;
        this.value = value;
        this.time = time;
        this.updateTime = updateTime;
        this.status = status;
    }

    public Transaction(String phone, String code, int value, String time) {
        this.id = -1;
        this.phone = phone;
        this.code = code;
        this.value = value;
        this.time = time;
        this.updateTime = "";
        this.status = false;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public int getValue() {
        return value;
    }

    public void setValue(int value) {
        this.value = value;
    }

    public String getTime() {
        return time;
    }

    public void setTime(String time) {
        this.time = time;
    }

    public String getUpdateTime() {
        return updateTime;
    }

    public void setUpdateTime(String updateTime) {
        this.updateTime = updateTime;
    }

    public boolean isStatus() {
        return status;
    }

    public void setStatus(boolean status) {
        this.status = status;
    }

    public String getDisplayValue()
    {
        String rt = "";
        int para = value;
        while (para/1000 > 0){
            rt += para < value?(para%1000 +","):(para%1000);
            para /= 1000;
        }
        rt += para;
        return rt;
    }
}

File: source/model/transaction.cpp
#include "transaction.h"

#include "comunication/qtandroidservice.h"

Transaction::Transaction(QObject *parent, QString phone,
                         QString code, int value, QString time):
    QObject(parent)
{
    this->m_phone = phone;
    this->m_code = code;
    this->m_value = value;
    this->m_time = time;
}

Transaction::Transaction(QObject *parent, int id, QString phone,
                         QString code, int value, QString time, QString updateTime, bool status):
    Transaction(parent,phone,code,value,time)
{
    this->m_id = id;
    this->m_updateTime = updateTime;
    this->m_status = status;
}

Transaction::~Transaction()
{

}

void Transaction::setId(int id)
{
    this->m_id = id;
    emit idChanged();
}
void Transaction::setPhone(QString phone)
{
    this->m_phone = phone;
    emit phoneChanged();
}
void Transaction::setCode(QString code)
{
    this->m_code = code;
    emit codeChanged();
}
void Transaction::setValue(int value)
{
    this->m_value = value;
    emit valueChanged();
}

void Transaction::setTime(QString time)
{
    this->m_time = time;
    emit timeChanged();
}
void Transaction::setUpdateTime(QString updateTime)
{
    this->m_updateTime = updateTime;
    emit updateTimeChanged();
}


void Transaction::setStatus(bool isUpdated)
{
    this->m_status = isUpdated;
    emit statusChanged();
}

QString Transaction::getDisplayValue()
{
    QString rt = "";
    int para = m_value;
    while (para/1000 > 0){
        log("para : "+QString::number(para));
        rt = (para < m_value?(Utility::getFullDigits(para%1000,3)+","):Utility::getFullDigits(para%1000,3))+rt;
        log("rt :"+rt);
        para /= 1000;
    }
    if(para != 0){
        rt = QString::number(para)+ "," + rt;
    }
    log(rt);
    return rt;
}

//private
void Transaction::log(QString msg)
{
    QtAndroidService::instance()->sendToService(msg);
}
